x/tokenfactory/types/denoms.go

```diff
@@ -0,0 +1,43 @@
+package types
+
+import (
+	"strings"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+)
+
+const (
+	ModuleDenomPrefix = "factory"
+)
+
+func GetTokenDenom(creator, nonce string) (string, error) {
+	denom := strings.Join([]string{"factory", creator, nonce}, "/")
+	return denom, sdk.ValidateDenom(denom)
+}
+
+func DeconstructDenom(denom string) (creator string, nonce string, err error) {
+	err = sdk.ValidateDenom(denom)
+	if err != nil {
+		return "", "", err
+	}
+
+	strParts := strings.Split(denom, "/")
+	if len(strParts) < 3 {
```

Comment:
Why are we allowing more slashes? Is it to allow tokens to be able to be sub-indexed?

Reply:
Yeah

Reply:
Or like imagine I want to issue a centralized bridged version of an asset on another chain whose denom name has a slash in it, then maybe I want my token name to be:
`factory/sunnyaddr/ibc/asdfasdf`